build/BuildSteps.cs

```diff
@@ -75,6 +75,43 @@ public static void DotnetPublish()
             }
         }
 
+        public static void FilterPowershellRuntimes()
+        {
+            var minifiedRuntimes = Settings.TargetRuntimes.Where(r => r.StartsWith(Settings.MinifiedVersionPrefix));
+            foreach (var runtime in Settings.TargetRuntimes.Except(minifiedRuntimes))
+            {
+                var powershellRuntimePath = Path.Combine(Settings.OutputDir, runtime, "workers", "powershell", "runtimes");
+
+                var allKnownPowershellRuntimes = Settings.ToolsRuntimeToPowershellRuntimes.Values.SelectMany(x => x).Distinct().ToList();
+                var allFoundPowershellRuntimes = Directory.GetDirectories(powershellRuntimePath).Select(Path.GetFileName).ToList();
+
+                // Check to make sure any new runtime is categorizied properly and all the expected runtimes are available
+                if (allFoundPowershellRuntimes.Count != allKnownPowershellRuntimes.Count || !allKnownPowershellRuntimes.All(allFoundPowershellRuntimes.Contains))
+                {
+                    throw new Exception($"Mismatch between classified Powershell runtimes and Powershell runtimes found. Classified runtimes are ${string.Join(", ", allKnownPowershellRuntimes)}." +
+                        $"{Environment.NewLine}Found runtimes are ${string.Join(", ", allFoundPowershellRuntimes)}");
+                }
+
+                // Delete all the runtimes that should not belong to the current runtime
+                var powershellForCurrentRuntime = Settings.ToolsRuntimeToPowershellRuntimes[runtime];
+                allFoundPowershellRuntimes.Except(powershellForCurrentRuntime).ToList().ForEach(r => Directory.Delete(Path.Combine(powershellRuntimePath, r), recursive: true));
+            }
+
+            // Small test to ensure we have all the right runtimes at the right places
+            foreach (var runtime in Settings.TargetRuntimes.Except(minifiedRuntimes))
```

Comment:
This may be unnecessary, or can be moved above in the first loop. But, I thought it's good to keep a sanity check just in case above code changes.

If people think otherwise, I would be ok without this block.